src/main/java/de/tum/in/www1/artemis/repository/ExamSessionRepository.java

```diff
@@ -16,48 +16,28 @@
 public interface ExamSessionRepository extends JpaRepository<ExamSession, Long> {
 
     @Query("""
-                SELECT count(es.id)
-                FROM ExamSession es
-                WHERE es.studentExam.id = :#{#studentExamId}
+            SELECT COUNT(es.id)
+            FROM ExamSession es
+            WHERE es.studentExam.id = :studentExamId
             """)
     long findExamSessionCountByStudentExamId(@Param("studentExamId") Long studentExamId);
 
     @Query("""
-                SELECT es
-                FROM ExamSession es
-                WHERE es.studentExam.exam.id = :examId
-                    AND es.id <> :#{#examSession.id}
-                    AND es.studentExam.id <> :#{#examSession.studentExam.id}
-                    AND es.ipAddress = :#{#examSession.ipAddress}
-                    AND es.browserFingerprintHash = :#{#examSession.browserFingerprintHash}
+            SELECT es
+            FROM ExamSession es
+            WHERE es.studentExam.exam.id = :examId
+                AND es.id <> :sessionId
+                AND es.studentExam.id <> :studentExamId
+                AND (:ipAddress IS NULL OR es.ipAddress = :ipAddress)
+                AND (:browserFingerprintHash IS NULL OR es.browserFingerprintHash = :browserFingerprintHash)
             """)
-    Set<ExamSession> findAllExamSessionsWithTheSameIpAddressAndBrowserFingerprintByExamIdAndExamSession(long examId, @Param("examSession") ExamSession examSession);
+    Set<ExamSession> findAllExamSessionsWithTheSameIpAddressAndBrowserFingerprintByExamIdAndExamSession(Long examId, Long sessionId, Long studentExamId, String ipAddress,
+            String browserFingerprintHash);
```

Comment:
Missing `@Param`.

Reply:
Well @Param is a whole other question.
It's only needed if your parameter names deviate from your query parameters. If we wanted to make it consistent throughout the system, we would first need to decide on that, and then I would also suggest doing this in other PRs.

Reply:
https://docs.artemis.cit.tum.de/dev/guidelines/server.html
![grafik](https://github.com/ls1intum/Artemis/assets/26540346/7dc67aa5-2ac9-47da-ab05-3ed4359459b8)

Reply:
Sigh, good to know. Then I have to recheck all other PRs 😂